### PR TITLE
feat(connect-explorer-nextra): scaffolding

### PR DESCRIPTION
**Pull request by martykan**
Scaffolding for new Connect Explorer using [Nextra](https://nextra.site)

## Related Issue

#11425

### CODE REVIEW

File: packages/connect-explorer-nextra/tailwind.config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+module.exports = {
+  content: [
+    "./pages/**/*.{js,jsx,ts,tsx,md,mdx}",
+    "./components/**/*.{js,jsx,ts,tsx,md,mdx}",
+  ],
+  theme: {
+    extend: {},
+  },
+  plugins: [],
+  darkMode: "class",
```
Comment (karliatto) [inline packages/connect-explorer-nextra/tailwind.config.js]: Are we going to use `tailwindcss` ?

File: packages/connect-explorer-nextra/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# NextJS ignore file
+.next/
+.env
```
Comment (karliatto) [inline packages/connect-explorer-nextra/.gitignore]: `.env` is included in root `.gitingore` so probably not needed here https://github.com/trezor/trezor-suite/blob/develop/.gitignore#L27

File: packages/utils/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "publishConfig": {
         "main": "lib/index"
     },
-    "main": "src/index",
+    "main": "src/index.ts",
```
Comment (karliatto) [inline packages/utils/package.json:20]: Is this part of connect-explorer-nexttra scaffolding? It looks not related to me.

File: packages/connect-explorer-nextra/tailwind.config.js (hunk 1 above)
Comment (martykan) [reply packages/connect-explorer-nextra/tailwind.config.js]: No probably we will use `styled-components` like Suite, but the default Nextra template is using it

File: packages/utils/package.json (hunk 3 above)
Comment (martykan) [reply packages/utils/package.json:20]: This is because without this change I was unable to import the `components` package, which depends on this package

File: packages/connect-explorer-nextra/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+yarn dev
+```
+
+## Building
+
+```bash
+yarn build
+```
+
+Static files will be generated in `out/` directory.
```
Comment (karliatto) [inline packages/connect-explorer-nextra/README.md]: It would be nice to change from `out/` to `build/` directory to have consistency.

**Comment by karliatto** [comment]
When I run the `dev` command it works but `build` throws and error like:

```
../components/src/components/Badge/Badge.tsx:36:12
Type error: Element implicitly has an 'any' type because expression of type '"transparent" 
 Property 'transparent' does not exist on type 'DefaultTheme'.
```

It looks like error coming from `@trezor/components`.

**Comment by karliatto** [comment]
Other thing, I think we should change the structure of the scaffolding with `src` directory that will include `components`, `pages` , `public` and `styles` to be something like:

```
connect-explorer-nextra/src/pages
connect-explorer-nextra/src/public
connect-explorer-nextra/src/components
...
connect-explorer-nextra/package.json
connect-explorer-nextra/README.md
connect-explorer-nextra/tsconfig.json
.....
```

**Comment by karliatto** [comment]
@martykan it is now deployed by GitHub Actions to https://dev.suite.sldev.cz/connect/feat/connect-explorer-nextra/

Edited:

Now we are deploying the dev connect explorer with nextra to: https://dev.suite.sldev.cz/connect-nextra/feat/connect-explorer-nextra/

And the old connect-explorer to: https://dev.suite.sldev.cz/connect/feat/connect-explorer-nextra/

File: packages/connect-explorer-nextra/README.md (hunk 4 above)
Comment (karliatto) [inline packages/connect-explorer-nextra/README.md]: This should be updated, we are now generating it to directory `build`.

File: packages/connect-explorer-nextra/next.config.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                'process.env.__TREZOR_CONNECT_SRC': JSON.stringify(
+                    process.env.__TREZOR_CONNECT_SRC,
+                ),
+                'process.env.COMMIT_HASH': JSON.stringify(commitHash),
+                'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV),
+            }),
+        );
+    },
+    typescript: {
+        // Problems with transpiling
```
Comment (karliatto) [inline packages/connect-explorer-nextra/next.config.js:38]: Do we still have those problems?

File: packages/connect-explorer-nextra/package.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    "name": "@trezor/connect-explorer-nextra",
+    "packageManager": "yarn@4.0.2",
+    "version": "1.0.0",
+    "private": true,
+    "scripts": {
+        "dev": "next",
+        "build": "next build && rm -rf build && mv out build",
+        "start": "next start",
+        "lint:js": "yarn g:eslint '**/*{.ts,.tsx}'",
+        "lint:styles": "npx stylelint './src/**/*{.ts,.tsx}' --cache --config ../../.stylelintrc",
```
Comment (karliatto) [inline packages/connect-explorer-nextra/package.json:11]: Why don't we add `stylelint` as dev dependency instead of using `npx`?

File: packages/connect-explorer-nextra/package.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
+{
+    "name": "@trezor/connect-explorer-nextra",
+    "packageManager": "yarn@4.0.2",
+    "version": "1.0.0",
+    "private": true,
+    "scripts": {
+        "dev": "next",
+        "build": "next build && rm -rf build && mv out build",
```
Comment (karliatto) [inline packages/connect-explorer-nextra/package.json:8]: I am surprise that next.js does not have a way to configure the output directory. But I guess it is like that.

File: packages/connect-explorer-nextra/public/login_dialog.png
Comment (karliatto) [inline packages/connect-explorer-nextra/public/login_dialog.png:1]: Regarding this file, is it used anywhere? To be hones it looks quite old to me I have never seen it. Or is it a new design that I am not aware of?

File: packages/connect-explorer-nextra/public/login_form.png
Comment (karliatto) [inline packages/connect-explorer-nextra/public/login_form.png:1]: Is this file used?

File: packages/connect-explorer-nextra/src/components/Param.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+interface ParamProps {
+    name: string;
+    type: string | React.ReactNode;
+    required?: boolean;
+    description?: string;
+    children?: React.ReactNode;
+}
+
+// mt-2 rounded-2xl bg-gradient-to-b from-neutral-100 to-transparent
```
Comment (karliatto) [inline packages/connect-explorer-nextra/src/components/Param.tsx]: Is this comment still needed?

File: packages/connect-explorer-nextra/src/components/ParamsTable.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    } else if (value[Kind] === 'Array') {
+        hasDescendants = complexObjects.includes(value.items[Kind]);
+    }
+    // Get the type name
+    const typeName = getTypeName(value, hasDescendants);
+
+    // Required can also be undefined (for example union)
+    let isRequired: boolean | undefined;
+    if (schema?.required?.includes(name)) isRequired = true;
+    else if (value[Optional] === 'Optional') isRequired = false;
```
Comment (karliatto) [inline packages/connect-explorer-nextra/src/components/ParamsTable.tsx]: I would really use brackets here. But up to you :)

File: packages/connect-explorer-nextra/src/components/fields/File.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+import type { Field } from '../../types';
+
+interface FileProps {
+    field: Field<any>;
+    disabled?: boolean;
+    onChange: (field: Field<any>, value: any) => any;
+}
+
+const File = ({ disabled, field, onChange }: FileProps) => {
+    const onFilesAdded = evt => {
```
Comment (karliatto) [inline packages/connect-explorer-nextra/src/components/fields/File.tsx:13]: I know this is copied from old code, but since we are here why don't we call it `event`? and the one bellow `onLoadEvent`?

File: packages/connect-explorer-nextra/package.json (hunk 6 above)
Comment (martykan) [reply packages/connect-explorer-nextra/package.json:11]: It's the same like in other packages

File: packages/connect-explorer-nextra/package.json (hunk 7 above)
Comment (martykan) [reply packages/connect-explorer-nextra/package.json:8]: I couldn't find the option. They have a `distDir` option but it's not the right one, it changes the `.next` folder

File: packages/connect-explorer-nextra/public/login_dialog.png
Comment (martykan) [reply packages/connect-explorer-nextra/public/login_dialog.png:1]: No, none of the `login_` images are used now

File: packages/connect-explorer-nextra/src/components/fields/File.tsx (hunk 10 above)
Comment (martykan) [reply packages/connect-explorer-nextra/src/components/fields/File.tsx:13]: Yes, this is just copied and unchanged for now

File: packages/connect-explorer-nextra/next.config.js (hunk 5 above)
Comment (martykan) [reply packages/connect-explorer-nextra/next.config.js:38]: Yes, because build:libs were removed in a lot of packages we are recompiling them, but there are some issues. 
So I toggled `ignoreBuildErrors: true` for now

File: packages/connect-explorer-nextra/package.json (hunk 6 above)
Comment (karliatto) [reply packages/connect-explorer-nextra/package.json:11]: Yes, I know but apparently there is not good reason to do that https://satoshilabs.slack.com/archives/G019WLX2P7B/p1710177041316439

Up to you.

File: packages/connect-explorer-nextra/package.json (hunk 7 above)
Comment (karliatto) [reply packages/connect-explorer-nextra/package.json:8]: Yes, you are right, but still surprising. Anyway it looks like your solution is the best one. :+1:

**Comment by martykan** [comment]
@SocketSecurity ignore-all

**Comment by karliatto** [comment]
It looks good, I would like to see all the tests passing in CI  https://gitlab.com/satoshilabs/trezor/trezor-suite/-/pipelines/1209952248

For now the `unchained.test` is being failing few runs.

**Review by karliatto — APPROVED** [review]
Let's go for it! :rocket: 

It should not break anything since it is separated from the connect-explorer that we use for tests.
